rfcs/0182-user-management.md

```diff
@@ -0,0 +1,306 @@
+---
```

Comment:
In order to keep the conversation more discoverable, I am copying and answering here:

@paperdigits - https://github.com/NixOS/rfcs/pull/182#issuecomment-2364690555

>So it'd be easier to rewrite from scratch than to bring home manager into the nixpkgs tree?

Not necessarily. The code does not need to be written _ex nihilo_.
In the worst case, ignoring all considerations over ethics, we could just merge the current state of home-manager and start working right now.
But I feel this is not correct.

> You say you see this competing with home manager, but we know defaults are king, and I fail to see how home manager would get new users if similar functionality is implemented in nixpkgs/nixos itself.

Assimilate HM would be veri nice.
Indeed I kept the previous proposal for assimilation [here](https://github.com/NixOS/rfcs/pull/182/commits/2e84b448aca02397ed528caaa63be7b906095294), but preliminary conversations on Discourse and Matrix led me to cogitate this approach.

Reply:
I definitely agree that everything does not need a rewrite, but I'd also like to highlight that many home-manager modules duplicate functionality that already exists in NixOS in some form (activation logic, systemd units, programs.* services.*). Large scale cleanup is going to be required to have a smooth transition that doesn't break in little places everywhere.